Files/UserControls/NavigationToolbar.xaml.cs

```diff
@@ -31,6 +31,9 @@ namespace Files.UserControls
 {
     public sealed partial class NavigationToolbar : UserControl, INavigationToolbar, INotifyPropertyChanged
     {
+        private MainPage mainPage => ((Window.Current.Content as Frame).Content as MainPage);
```

Comment:
I am not sure the toolbar should be accessing MainPage, what do you think?

Reply:
@yaichenbaum Do you have any alternative?

Reply:
Can we use a dependency property instead?

Reply:
@yaichenbaum Where? Would it ever make sense to change this reference from where the control is used?

Reply:
@duke7553 I don't think so, however, if we want to view this as a decoupled control, we shouldn't have the reference here.

Reply:
@yaichenbaum I'm not aware of any alternative based on how things are designed.

Reply:
For the record, we'll keep this reference until we move vertical tabs out of the control and into MainPage with a later PR. It's a temporary solution that we can't do any better with right now. Also, take a look at PaneHolderPage. I removed the reference from there completely.